packages/venia-ui/lib/RootComponents/CMS/cms.js

```diff
@@ -1,8 +1,41 @@
-import React, { Component } from 'react';
-import CategoryList from '../../components/CategoryList';
+import React, { useEffect } from 'react';
+import { useQuery } from '@magento/peregrine';
+import cmsPageQuery from '../../queries/getCmsPage.graphql';
+import { fullPageLoadingIndicator } from '../../components/LoadingIndicator';
+import RichContent from '../../components/RichContent';
 
-export default class CMS extends Component {
-    render() {
-        return <CategoryList title="Shop by category" id={2} />;
+const CMSPage = props => {
+    const { id } = props;
+    const [queryResult, queryApi] = useQuery(cmsPageQuery);
+    const { data, error, loading } = queryResult;
+    const { runQuery, setLoading } = queryApi;
+
+    useEffect(() => {
+        setLoading(true);
+        runQuery({
+            variables: {
+                id: Number(id),
+                onServer: false
+            }
+        });
+    }, [id, runQuery, setLoading]);
+
+    if (error) {
+        return <div>Page Fetch Error</div>;
```

Comment:
This will consume the error. We should either show a toast or log it to console. To add a toast look at some of the other examples. Search for `addToast`.

Reply:
Classic @zetlen contributing bad code...

I've integrated the error handling you have in your other RootComponents into here, `addToast` doesn't seem to be widely used for application errors?

```
        if (process.env.NODE_ENV !== 'production') {
            console.error(error);
        }
```